lib/segment/src/data_types/order_by.rs

```diff
@@ -109,22 +109,28 @@ impl OrderBy {
         new_payload
     }
 
-    pub fn remove_order_value_from_payload(&self, payload: Option<&mut Payload>) -> f64 {
+    pub fn get_order_value_from_payload(&self, payload: Option<&Payload>) -> OrderingValue {
         payload
-            .and_then(|payload| payload.0.remove(INTERNAL_KEY_OF_ORDER_BY_VALUE))
-            .and_then(|v| v.as_f64())
+            .and_then(|payload| payload.0.get(INTERNAL_KEY_OF_ORDER_BY_VALUE))
+            .and_then(|v| OrderingValue::try_from(v.clone()).ok())
             .unwrap_or_else(|| match self.direction() {
-                Direction::Asc => std::f64::MAX,
-                Direction::Desc => std::f64::MIN,
+                Direction::Asc => OrderingValue::MAX,
+                Direction::Desc => OrderingValue::MIN,
             })
     }
 }
 
+#[derive(Debug)]
 pub enum OrderingValue {
     Float(FloatPayloadType),
     Int(IntPayloadType),
 }
 
+impl OrderingValue {
+    const MAX: Self = Self::Float(f64::NAN);
```

Comment:
NAN -> MAX?

Reply:
According to out `Ord` implementation it is the highest value possible